### ISSUE TITLE
SQLite alternative?

### ISSUE THREAD

**Issue by Soullivaneuh** [luxifer/doctrine-functions#23]
SQLite is frequently used instead of MySQL for unit testing.

But some functions differ.

Would you be interested by Pull Requests for SQLite alternatives?

This is a quite easy implement, for example with `YEAR`: 

```php
public function getSql(SqlWalker $sqlWalker)
{
    if ($sqlWalker->getConnection()->getDriver() instanceof AbstractSQLiteDriver) {
        return "DATE('%Y', {$this->dateExpression->dispatch($sqlWalker)})";
    }

    return 'YEAR('.$this->dateExpression->dispatch($sqlWalker).')';
}
```

What do you think?

**Comment by luxifer** [comment]
Yes!!!! If the result of the different function are equal, it's OK for me

**Comment by Soullivaneuh** [comment]
@luxifer I think the tests should be rethink in this case, to test the result of a query on real databases, right?

**Comment by luxifer** [comment]
I don't think it's useful to test that as it will introduce a lot of overhead in the test suite. As long as the MySQL and SQlite doc specify the same value for the return, it's OK for me
